### ISSUE TITLE
[Wordpress / Woocommerce] Landing Pages - Create structure and main functionality

### ISSUE THREAD

**Issue by eduardogomez97** [doofinder/doofinder-woocommerce#212]
Create a controller called landing.php in the plugin that receives two parameters through the URL. Hashid and Slug.

The controller has to load the default template that WP has and make a call to doomanager passing the hashid and the slug. 

With this information we make a call to the doomanager endpoint to get the landing page information. /plugins/landing/' . $this->hashid . '/' . $slug;

This returns the H1, the content of the blocks and SEO information.

In the content of the blocks we have the query for doophoenix to get the custom results that we want to display on the screen.

Important the template we generate has to have a basic HTML structure and WP generic classes to not interfere with the theme styles.

The structure would be:

doofinder-for-woocommerce/views/landing.php → Template.
doofinder-for-woocommerce/include/class-landing.php → Controller
doofinder-for-woocommerce/include/api/class-landing-api.php → Client.

**Comment by eduardogomez97** [comment]
Regarding this task.

**Landing context.**

- [ ] If we access doofinder-for-woocommerce/views/landing.php and pass the parameters slug and hashid it works. 
- [ ] If the hashid is missing we try to get the hashid from the current lang. 
- [ ] If the slug is missing we have to get it directly from the url /df/loquesea.
- [ ] In any other case that the parameters could not be obtained the error message is displayed or redirects to page 404 and returns no index.
- [ ] The call to doomanager is made to fetch the landing data.
- [ ] With the query that comes in each block we make a call to doophoenix to get the necessary ids.
- [ ] Function that returns the HTML when the call to doomanager gives an error.
- [ ] Function that returns the HTML when the landing data is fetched.
- [ ] In this last function, when the products have to be painted a function is called that builds the structure of each block.
- [ ] Each block has to make a call to get the products it has to paint and make the corresponding structure.

**Template landing**

- [ ] Get WP scripts and styles.
- [ ] Get WP header.
- [ ] Assign the language to the page.
- [ ] Get parameters get and send it to the get_landing_info function.
- [ ] Set meta parameters if there is no error.
- [ ] Get the corresponding HTML template if there is an error or not.
- [ ] Get WP footer.

**API Landing**

- [ ] API Builder
- [ ] Call to the doomanager endpoint to get the landing data.
- [ ] Build the sendRequest.
- [ ] Call to doophoenix to get the ids thanks to the query (I have a mock right now to test).

I have some doubts about the doophoenix end-point and the query I have to do so I currently have it mocketed. When I come back from vacation I will discuss it with a fellow AZKB member.

**Comment by CristobalDolz** [comment]
@eduardogomez97 para cuando vuelvas de las vacaciones. quería preguntarte si las conversion pages que se van a crear desde doomanager luego se crean tambien en la sección de Paginas del Wordpress como si fueran páginas nativas



![Image](https://github.com/doofinder/doofinder-woocommerce/assets/106097983/e3b3b53d-f24b-4da2-93a5-f0540c850e30)

**Comment by eduardogomez97** [comment]
Buenas @Cristobal Dolz Oltra ***@***.***>

Pues siguiendonos del modelo de referencia de Prestashop no se crearía una
página estática. Es una página dinámica que se construye en el momento con
los datos que recoge.

Cualquier cosa me puedes comentar, que ya estoy de vuelta.

**Comment by eduardogomez97** [comment]
Earlier in the week I was doing the final tests and we saw a bug with the woocommerce shortcode system. So I had to find another way to do it using the templates that the client has activated, it has been possible to do it after an investigation and I have found several errors. However right now changing the theme our conversion page adapts.

I am improving the code and testing to solve cases that I have been finding, as well as adding the relevant documentation. A file will be created with the requirements to have the conversion pages activated and a guide of how it works.
